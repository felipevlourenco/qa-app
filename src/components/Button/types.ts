import { ReactNode } from 'react'

export interface ButtonProp {
  children: ReactNode
  onClick?: () => void
  disabled?: boolean
  variant?: 'primary' | 'secondary'
  type?: 'button' | 'submit' | 'reset'
}
