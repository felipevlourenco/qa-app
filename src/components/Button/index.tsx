import React from 'react'

const Button = () => {
  return (
    <button type="button">
      <span>test button</span>
    </button>
  )
}

export default Button
