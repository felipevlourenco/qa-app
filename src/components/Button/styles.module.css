.button {
  font-size: 1rem;
  padding: 0.5rem;
  border: 0;
  border-radius: 4px;
}

.button:hover {
  filter: brightness(1.2);
}

.button:disabled {
  filter: brightness(1.0);
  background-color: #ccc;
}

.primary {
  color: #ffffff;
  background-color: #318ed5;
}

.secondary {
  color: #ffffff;
  background-color: #e26a6a;
}