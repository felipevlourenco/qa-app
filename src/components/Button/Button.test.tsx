import { fireEvent, render } from '@testing-library/react'
import Button from 'components/Button'

const defaultProp = {
  onClick: jest.fn
}

describe('components/Button', () => {
  it('should render', () => {
    const { container } = render(<Button {...defaultProp}>Button</Button>)
    expect(container).toMatchSnapshot()
  })

  it('should call onClick callback', () => {
    const onClick = jest.fn()
    const { getByText } = render(<Button onClick={onClick}>Button</Button>)
    fireEvent.click(getByText('Button'))
    expect(onClick).toBeCalledTimes(1)
  })

  it('should not call onClick callback if button is disable', () => {
    const onClick = jest.fn()
    const { getByText } = render(
      <Button onClick={onClick} disabled>
        Button
      </Button>
    )
    fireEvent.click(getByText('Button'))
    expect(onClick).toBeCalledTimes(0)
  })
})
