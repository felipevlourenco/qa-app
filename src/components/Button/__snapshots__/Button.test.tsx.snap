// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button should render 1`] = `
<div>
  <button
    class="button primary"
    type="button"
  >
    Button
  </button>
</div>
`;
